app/components/traffic-signal-common/shape-manager.gts

```diff
@@ -0,0 +1,575 @@
+    ) {
+      for (const shape of this.shapesConverted.value) {
+        await shape?.convertToNewUnit(this.unitChange, this.store);
+      }
+      this.shapesConverted.retry();
+      this.cardEditing = false;
+    }
+  };
+
+  closeShapeChangeConfirmation = () => {
+    this.isShapeChangeConfirmationOpen = false;
+  };
+
+  changeShape = async () => {
+    let shapes = [...(await this.args.trafficSignal.shapes)];
+    for (const shape of shapes) {
+      await shape.destroyWithRelations();
+    }
+    const shapeClass = SHAPES[this.shapeChange?.uri as keyof typeof SHAPES];
+    const shape = (await shapeClass.createShape(
+      this.selectedUnit as Unit,
+      this.store,
+      this.args.trafficSignal,
+    )) as Shape;
+    for (const dimension of shapeClass.headers(this.intl)) {
+      const dimensionProperty =
+        shape[dimension.value as keyof typeof DIMENSIONS];
+      if (dimensionProperty) {
+        dimensionProperty.value = 0;
+      }
+    }
+    await shape.validateAndsave(this.store);
+    this.args.trafficSignal.set('defaultShape', undefined);
+    this.args.trafficSignal.set('shapes', [shape.shape]);
+    await this.store.request(saveRecord(this.args.trafficSignal));
+    this.shapesConverted.retry();
+    this.cardEditing = false;
+    this.closeShapeChangeConfirmation();
+  };
+
+  saveShape = async () => {
+    const saved = await this.shapeToEdit?.validateAndsave(this.store);
+    if (saved) {
+      const shape = this.shapeToEdit?.shape as TribontShape;
+      console.log(this.convertToNewDefaultShape);
+      if (this.convertToNewDefaultShape) {
+        this.args.trafficSignal.set('defaultShape', shape);
+      } else if (this.shapeToEdit?.id === this.defaultShape.value?.id) {
+        this.args.trafficSignal.set('defaultShape', undefined);
+      }
+      await this.store.request(saveRecord(this.args.trafficSignal));
+      await this.shapesConverted.retry();
+      await this.closeEditShapeModal();
+    }
+  };
+
+  cancelCard = () => {
+    this.cardEditing = false;
+    this.unitChange = undefined;
+    this.shapeChange = undefined;
+  };
+
+  getStringValue(shape: Shape, dimension: keyof typeof DIMENSIONS) {
+    if (!shape[dimension]) return '';
+    return shapeDimensionToText(shape[dimension]);
+  }
+
+  setShapeClassifications = (classification: ShapeClassification) => {
+    this.shapeChange = classification;
+  };
+  setUnit = (unit: Unit) => {
+    this.unitChange = unit;
+  };
+
+  startDeleteShapeFlow = (shape: Shape) => {
+    this.shapeToDelete = shape;
+    this.isDeleteConfirmationOpen = true;
+  };
+
+  closeDeleteConfirmation = () => {
+    this.shapeToDelete = undefined;
+    this.isDeleteConfirmationOpen = false;
+  };
+
+  removeShape = async () => {
+    const shape = this.shapeToDelete;
+    if (!shape) return;
+    const shapes = await this.args.trafficSignal.shapes;
+    removeItem(shapes, shape.shape);
+    await this.store.request(saveRecord(this.args.trafficSignal));
```

Comment:
Why are these 3 lines necessary if we are removing the shape? It also fetches all the shapes unpaginated...